### ISSUE TITLE
Unable to read latin-1 encoded SAS Datasets using encoding option

### ISSUE THREAD

**Issue by flying-fluid-four** [Roche/pyreadstat#117]
I have a SAS dataset with Latin-1 encoding as follows::
![final1](https://user-images.githubusercontent.com/80445989/110746134-5575cf00-820a-11eb-9fce-e7a7f3b6efba.jpg)

I am trying to read this using pyreadstat with encoding specified as Latin1 with a failure message as follows:
![final2](https://user-images.githubusercontent.com/80445989/110746164-632b5480-820a-11eb-93d8-a85ccae6f981.jpg)

**Comment by flying-fluid-four** [comment]
Additional info: the same file is being read successfully by pandas read_sas method as well as by converting the encoding to utf-8(within SAS) and then reading by pyreadstat using encoding = 'utf-8' option.

**Comment by ofajardo** [comment]
Thanks for the report. Please provide the file. 

If you can't I am afraid that I won't be able to debug it. 

Make sure you are using the latest version of pyreadstat. There was a bug for this kind of thing in the past.

Have you tried to read it without the encoding argument? That will let the system try to guess the encoding. You can also try other encodings from here: https://gist.github.com/hakre/4188459, notice that all of them are in capitals.

**Comment by flying-fluid-four** [comment]
Hi, yes I tried removing the encoding option and as well as capitalizing the "LATIN1". Not sure which other encoding option I can use from the list. Do you recommend any other from that list?

**Comment by flying-fluid-four** [comment]
Also, unfortunately unable to provide the SAS File as I don't have a SAS License on my personal computer :/

**Comment by ofajardo** [comment]
version of pyreadstat:

```
import pyreadstat
print(pyreadstat.__version__)
```

you can also do pip list or conda list and check.

Otherwise no other recommendation I am afraid. If no file can't do much more. I don't think you need a SAS license in order to share a SAS file (maybe I am wrong), the important point is that the SAS file does not contain any sensitive or copyrighted information.

**Comment by flying-fluid-four** [comment]
Ok, so I am running version 1.0.5

And yes, data is copyrighted and I don't have license on my local machine to create non sensitive SAS Dataset. I'll try to see if I can find a latin-1 encoded file on the web somewhere,

**Comment by ofajardo** [comment]
OK, please update to version 1.1.0 and try again first:

```
pip install pyreadstat --upgrade
```

or if you installed with conda:

```
conda update pyreadstat
```

**Comment by flying-fluid-four** [comment]
Ok, will try tomorrow and let you know! Thanks.

**Comment by ofajardo** [comment]
Yep, maybe upgrading solves everything. Otherwise I remember that SAS offered some free licenses you could use for personal use or for learning SAS only (not for real work), they are called SAS University I think or something like that. I guess this use may qualify as personal use?

**Comment by flying-fluid-four** [comment]
Found a sample dataset on the SAS WebSite
```
import pyreadstat  
print('Version of pyreadstat is {}'.format(pyreadstat.__version__))
test = pyreadstat.read_sas7bdat('C:\\Users\\SeanP\\Desktop\\SASds\\DCSKINPRODUCT.sas7bdat')
``
Version of pyreadstat is 1.1.0
Traceback (most recent call last):

  File "<ipython-input-11-6cc5d8df1b61>", line 3, in <module>
    test = pyreadstat.read_sas7bdat('C:\\Users\\SeanP\\Desktop\\SASds\\DCSKINPRODUCT.sas7bdat')

  File "pyreadstat\pyreadstat.pyx", line 116, in pyreadstat.pyreadstat.read_sas7bdat

  File "pyreadstat\_readstat_parser.pyx", line 1034, in pyreadstat._readstat_parser.run_conversion

  File "pyreadstat\_readstat_parser.pyx", line 845, in pyreadstat._readstat_parser.run_readstat_parser

  File "pyreadstat\_readstat_parser.pyx", line 775, in pyreadstat._readstat_parser.check_exit_status

ReadstatError: Invalid file, or file has unsupported features

**Comment by ofajardo** [comment]
OK, thanks a lot for the example file. I tried an old version of pyreadstat (0.2.7) and there it works. The error is coming from the C library 'Readstat' so it means some change in their code make this to break. And btw, the encoding of this file (looking at it with proc contents) is utf-8, so it is not an encoding issue. I hope that the issue you are seeing in this sample file is the same as in your real file (it may be two different things).

In any case, I will report this to Readstat for fixing.

**Comment by ofajardo** [comment]
fixed in dev for the sample file provided. Hope it also fixes your problematic private file. I will close this once I make a new release (waiting for a readstat release right now)

**Comment by ofajardo** [comment]
the issue should be now fixed in the new version 1.1.1
